tools/sof-kernel-log-check.sh

```diff
@@ -127,10 +127,10 @@
 # than deleting it.
 
 
-begin_line=${1:-1}
+begin_timestamp=${1:-0}
```

Comment:
Because I drop the usage for line number now. And with my comments below
`# confirm begin_timestamp is in UNIX timestamp format, otherwise use dmesg`

We will always fallback to use demsg if no checkpoint is provided or wrong format checkpoint is provided

Reply:
> We will always fallback to use demsg if no checkpoint is provided or wrong format checkpoint is provided

The only user of this script is ourselves, so to reduce complexity a lot let's just fix the user instead of providing a fallback. Fail here when some test does not provide a valid checkpoint and then calls this script, which seems very wrong!? Let's fix all these broken tests and make them consistent _before_ switching the checkpoint to journalctl

Reply:
I thought we have an agreement in https://github.com/thesofproject/sof-test/pull/468#issuecomment-718311303
> @marc-hb Thanks for the tips, I am also think about how to spilt the patches. Please help to check my plan:
> 
> 1. replace LINE with TIMESTAMP, log check with no `--priority` but still with `err_str`
> 2. quoting clean up for the `$KERNEL_LAST_TIMESTAMP`
> 3. remove `err_str` and use `--priority`, add new `ignore_str`
> 4. clean up test case to remove duplicate kernel log check.

Now we have done half of the step **1** by changing the LINE into CHECKPOINT in https://github.com/thesofproject/sof-test/pull/499. Now I am not sure if I can do clean up in step **4** before we change to use journalctl with use `--priority`, it would bring more unrelated error in our test now.


Or we can do something more simpler, merge a big PR https://github.com/thesofproject/sof-test/pull/468 I have already made and tested and fix bugs if we find any.

Any suggestions here @marc-hb  @aiChaoSONG @plbossart @mengdonglin 
I am not sure what we are reviewing here. I do not see any feedback about codes or implementation. I only see how to spilt patches. For such big implementation change, I am not sure if small steps can make thing easier because I need to do some un-needed migration clean up for each small change. EG: @marc-hb The **fallback** here is one of those things I do not need to have in big PR. But I have to keep it here for migration.

If we do not make progress before my vacation next week. I will use my super power to merge all the change in one PR.